src/activity_handlers.cpp

```diff
@@ -2538,17 +2539,39 @@ void activity_handlers::craft_do_turn( player_activity *act, player *p )
 {
     const recipe &rec = recipe_id( act->name ).obj();
     float crafting_speed = p->crafting_speed_multiplier( rec, true );
-    if( crafting_speed <= 0.0f ) {
-        if( p->lighting_craft_speed_multiplier( rec ) <= 0.0f ) {
-            p->add_msg_if_player( m_bad, _( "You can no longer see well enough to keep crafting." ) );
-        } else {
-            p->add_msg_if_player( m_bad, _( "You are too frustrated to continue and just give up." ) );
+    bool changed_light = false;
+
+    if( p->lighting_craft_speed_multiplier( rec ) <= 0.0f ) {
+        auto &&v = g->m.veh_at( p->pos() );
+        if( v.has_value() &&
+            ( act->moves_total == act->moves_left ||
+              query_yn( "It's getting too dark to craft.  Would turning on lights on help?" ) ) ) {
+            /* If the player is on a tile with a dome light, we only need to turn on the dome lights. */
+            vehicle *veh = &( v->vehicle() );
+            if( veh->get_parts( p->pos(), "CONTROLS", false, false ).size() > 0 ||
+                veh->get_parts( p->pos(), "CTRL_ELECTRONIC", false, false ).size() > 0 ) {
+                changed_light = veh->turn_on_internal_lights( true );  // Turn on the dome lights
+            } else {
+                changed_light = veh->turn_on_internal_lights(); // Turn on all other internal lights (not dome)
+            }
+            p->mod_moves( -300 );
+        }
+    } else {
+        if( crafting_speed <= 0.0f ) {
+            if( p->lighting_craft_speed_multiplier( rec ) <= 0.0f ) {
```

Comment:
This can never be true because of L@2544

Reply:
Fixed.